bigquery_etl/public_data/publish_json.py

```diff
@@ -228,7 +230,10 @@ def _gcp_convert_ndjson_to_json(self, gcs_path):
 
     def _write_results_to_temp_table(self):
         """Write the query results to a temporary table and return the table name."""
-        table_date = self.date.replace("-", "")
+        table_date = self.date.replace("-", "") + "".join(
+            random.choice(string.ascii_lowercase) for i in range(12)
+        )
+
         self.temp_table = (
             f"{self.project_id}.tmp.{self.table}_{self.version}_{table_date}_temp"
         )
```

Comment:
The temporary tables get deleted in https://github.com/mozilla/bigquery-etl/blob/1b6128ff5aee11c12ee440cda3567083b05d7aee/bigquery_etl/public_data/publish_json.py#L79

I think the problem here was that there were multiple integration test instances running in parallel, trying to create the same temporary table and failing when it already existed due to another test having created it.

Reply:
based on https://stackoverflow.com/a/1481512, that method is not guaranteed to be called at all, and `bq ls bigquery-etl-integration-test:tmp` shows that tables created in CI are not being cleaned up.

I think this needs to delete the temp table in a `finally` block somewhere, to ensure that it actually runs.